lib/iris/cube.py

```diff
@@ -1835,6 +1835,10 @@ def summary(self, shorten=False, name_padding=35):
 
         nameunit = '{name} / ({units})'.format(name=self.name(),
                                                units=self.units)
+        # If all unknown and a STASH attribute exists, use it.
+        if (nameunit == 'unknown / (unknown)' and
+           'STASH' in self.attributes):
+            nameunit = '{}'.format(self.attributes.get('STASH'))
```

Comment:
`self.attributes['STASH']` ... if you wanna, as you've already checked that the `STASH` key is available.